# Proyecto 
## Reacción de síntesis de Metanotiol en un reactor PFR

Enunciado del Problema

In [77]:
#Librería para evaluar la integral de la fórmula del reactor PFR
from sympy import *
init_printing()

La siguiente reacción se verifica a 800 °C y a una presión constante de 1 atm. En un reactor de flujo de tipo tapón:
  
  

<p><center> $CH_{4} + S \rightarrow CH_{3} SH$</center></p>

La velocidad de la reacción a 800°C está dada por la expresión:

<p><center> $-r_{A} = k * C_{A} * C_{B}$</center></p>

Donde el valor de k a 800 °C es 10 L/mol·s. La alimentación de gas contiene 25% en mol de CH4, 25% mol de N2 y 50% de vapores de azufre.

- a) Calcular cual debe ser el tiempo de residencia para que haya un 60% de conversión de CH4 a CH3SH.
- b) Si la alimentación al reactor es de 10 mol/s en total, ¿Cuál debe ser el volumen del reactor para un 60% de conversión?
- c) ¿Cuál es el espacio velocidad medio en condiciones de operación del reactor?

### <font color="purple">Tabla Estequimétrica</font>


| Compuesto |Nomenclatura|--------Inicial-----------|----------Cambio--------------|----------------Salida-------------------|
|-----------|:-----------|:------------------------:|:----------------------------:|:---------------------------------------:|
| Metano    |A           |$F_{Ao}$                  |$-F_{Ao}(X)$                  |$F_{A}=F_{Ao}(1-X)$                      |
| Azufre    |B           |$F_{Bo}=\Phi_{B}F_{Ao}$   |$-(\frac{b}{a})F_{Ao}(X)$     |$F_{B}=F_{Ao}(\Phi_{B}-\frac{b}{a}X)$    |
| Metanotiol|C           |$0$                       |$+(\frac{c}{a})F_{Ao}(X)$     |$F_{C}=F_{Ao}(\frac{c}{a}-X)$            | 
| Nitrógeno |I           |$F_{Io}=\Phi_{I}F_{Ao}$   |$-F_{Ao}(X)$                  |$F_{I}=F_{Ao}(\Phi_{I})$                 |

### <font color="purple">Datos conocidos</font>

- Sitema en fase gaseosa
- Reacción no reversible
- Reactor PFR
- Sistema isotérmico
- Sistema isobárico


In [78]:
To = T = 800+273.15 #K
Po = P = 1          #atm 
K  = 10             #L/mol*s
R  = 0.082          #L*atm/mol*K
YAo= 0.25
YBo= 0.50
YIo= 0.25

#Cálculo de ƟB
ƟB = YBo/YAo
#Cálculo de ƟB
ƟI = YIo/YAo
print(ƟB)

2.0


In [79]:
print(ƟI)            #adimensional

1.0


####  <font color="purple"> - Cálculo de concentració inicial de A </font>
$C_{Ao}= (Y_{A})*(C_{To})=(Y_{A})*\Big(\frac{P}{R* To}\Big)$

In [80]:
CAo = YAo*(P/(R*T))
print(CAo)           #mol/dm^3

0.002840963973167663


####  <p><font color="purple"> - Cálculo de parámetro de relación de números de moles </font></p>

<p><center> $CH_{4} + S \rightarrow CH_{3} SH$</center></p> 
<center> $a A + b B \rightarrow c C$</center> 

$\delta=\Big(\frac{c}{a}-\frac{b}{a}-1\Big)$

- $ a=1 $
- $ b=1 $
- $ c=1 $

In [81]:
a = 1
b = 1
c = 1
ẟ = (c/a-b/a-a/a)
print(ẟ)             #adimensional

-1.0


####  <p><font color="purple"> - Cálculo de parámetro ε</font></p>
$\varepsilon = Y_{A} *\delta$

In [82]:
ε= YAo*ẟ
print(ε)              #adimensional

-0.25


####  <p><font color="purple"> - Ley de velocidad en términos de $C_{Ao}$</font></p>
Primero se toma en cuenta que el sistema en fase gaseosa por lo tanto:
<p>$\upsilon=\upsilon_{o}*(1+\varepsilon *x)*(\frac{P_{o}}{P})*(\frac{T_{o}}{T})$ sí $P=P_{o},T=T_{o}$ entonces </p>

<p>$\upsilon=\upsilon_{o}*(1+\varepsilon *x)$</p>

Siendo la ley de velocidad representada como:
<p>$-r_{A} = k * C_{A} * C_{B}$</p>

<p>$-r_{A} = k * \Big(\frac{F_{A}}{\upsilon}\Big) * \Big(\frac{F_{B}}{\upsilon}\Big) $</p>

<p>$-r_{A} = k * \Big\{\Big(\frac{F_{Ao}}{\upsilon}\Big) * \frac{(1-x)}{\upsilon_{o}*(1+\varepsilon x)}\Big\}* \Big\{\Big(\frac{F_{Ao}}{\upsilon}\Big) * \frac{(\Phi_{B}-\frac{b}{a}x)}{\upsilon_{o}*(1+\varepsilon x)}\Big\} $</p>

<p>$-r_{A} = k *\Big(\frac{C_{Ao}*(1-x)}{ (1 +\varepsilon x)}\Big)*\Big(\frac{C_{Ao}(2-x)}{(1+\varepsilon x)}\Big)$</p>

<p><font color="blue">$-r_{A} = k *\Big({C_{Ao}\Big)^2 *\Big\{\frac{(1-x)(2-x)}{(1+\varepsilon x)^2}}\Big\}$</font></p>

<p><font color="purple">- a) Calcular cual debe ser el tiempo de residencia para que haya un 60% de conversión de CH4 a CH3SH.</font></p>

Conversión de 60%, $x = 0.60$

$t = N_{Ao}\int\limits_0^x \frac{dx}{(-r_{A})*(V)} = \frac{N_{Ao}}{V}\int\limits_0^x \frac{dx}{(-r_{A})}=C_{Ao}\int\limits_0^x \frac{dx}{(-r_{A})*(V)}$

$t = \frac{C_{Ao}}{k*(C_{Ao})^2}\int\limits_0^x \frac{(1+\varepsilon x)^2}{(1-x)(2-x)}dx=\frac{1}{k*C_{Ao}}\int\limits_0^x \frac{(1-0.25 x)^2}{(1-x)(2-x)}$

$t = \frac{1}{k*C_{Ao}}\int\limits_0^x\frac{(1-0.25 x)^2}{(1-x)(2-x)}$ cuando $x=0.60$

In [103]:
x = symbols("x")
f=(((1+(ε*x)))**2)/((1-x)*(ƟB-x))
t1= integrate(f,(x,0,0.6) )
t0=0.463744800694529

In [104]:
print(t0)               #dentro de la integral

0.463744800694529


In [94]:
t_1=(1/(K*CAo))
print(t_1)               #fuera de la integral

35.19932000000001


In [114]:
t=t0*t_1
print(t)                #segundos

16.32350163798295


<p><font color="blue">-$RESPUESTA:$ El tiempo de residencia necesario en un Reactor PFR para un conversión del 60% de metano en metanotiol es de 16.33 segundos.</font></p>

<p><font color="purple">- b) Si la alimentación al reactor es de 10 mol/s en total, ¿Cuál debe ser el volumen del reactor para un 60% de conversión?</font></p>

<p>$F_{To}= 10 \frac{mol}{s}$</p>

####  <p><font color="purple"> - Cálculo de $F_{Ao}$ </font></p>
$F_{Ao}= F_{To} * Y_{Ao}$



In [109]:
FTo= 10                  #mol/s
FAo=FTo*YAo
print(FAo)               #mol/s

2.5


####  <p><font color="purple"> - Cálculo de volumen a las condiciones dadas </font></p>

Se sabe que tiempo de residencia se puede dar de la siguiente manera:
<p>$V_{o}= F_{Ao} / C_{Ao}$</p>
<p>$V= V_{o} * t$</p>

In [110]:
Vo=FAo/CAo                #dm^3/s
V=Vo*t
print(V)                  #dm^3

14364.403941897153


<p><font color="blue">-$RESPUESTA:$ El volumen necesario en un Reactor PFR para un conversión del 60% es de $14,364.40  dm^{3}$.</font></p>

<p><font color="purple">- c) ¿Cuál es el espacio velocidad medio en condiciones de operación del reactor?</font></p>

Se sabe que espacio velocidad medio esta dado por:
</p>$\tau=\frac{1}{t}$</p>


In [111]:
τ=1/t
print(τ)                  #s^-1

0.061261365494834304


<p><font color="blue">-$RESPUESTA:$ En condiciones de operación del Reactor PFR es de $0.0613  s^{-1}$.</font></p>